### ISSUE TITLE
Remove first empty line from magit buffer

### ISSUE THREAD

**Issue by argshook** [jreybert/vimagit#123]
When i open magit buffer, the first line is empty:

<img width="202" alt="screenshot 2017-03-01 14 35 00" src="https://cloud.githubusercontent.com/assets/4284659/23460139/65565972-fe8c-11e6-8c5c-40bef7ce303c.png">

i would like to remove it and have contents start at the very top.

wasn't able to find in code where this line is added. if anyone can point me to it, i'd gladly PR this.

**Comment by jreybert** [comment]
this empty line is here because all text in magit buffer in print with `put =`. put append text to the next line.

I tried to play around with `append()` or `setline()`, but there is still a bug (it breaks smart cursor jump when staging).

Anyway, the function makes a lot of thing instead of a simple `put`, I don't know the performance impact. You can try it out with the branch `dev/issue_123`

**Comment by jreybert** [comment]
Actually, I found the last bug. It seems to work, but I am still not sure I will integrate it.

**Comment by argshook** [comment]
wow this is great. i'll try it out tomorrow at work with more/bigger commits to see if performance got any worse. though i doubt it, no functions look expensive to my untrained eye.

i think you could also benefit from this by refactoring some code so that it doesn't call `s:print()` for each segment but instead create a list and print once, e.g instead of:

```vim
call s:print('')
call s:print('Press ? to display help')
call s:print('')
call s:print('')
```

do:

```vim
call s:print(['', 'Press ? to display help', '', ''])
```

same can go for entire sections.
you already made `s:print` to support lists so this should work too?

**Comment by jreybert** [comment]
Merged in master and present in release 1.7.2
